### ISSUE TITLE
How to control thermald throttling?

### ISSUE THREAD

**Issue by hxss** [intel/thermal_daemon#146]
Hello.
I'm use [thermald](https://www.archlinux.org/packages/community/x86_64/thermald/) on Arch Linux(4.14.13-1-zen). My laptop - [Lenovo Yoga 710-11isk](https://www3.lenovo.com/us/en/laptops/yoga/700-series/Lenovo-YOGA-710-11ISK/p/88YG7000723#tab-techspec) with [Intel® Core™ m5-6Y54](https://ark.intel.com/products/88202/Intel-Core-m5-6Y54-Processor-4M-Cache-up-to-2_70-GHz) Processor(only passive cooling). I was trying to configure thermald for comfortable Linux usage on this laptop, but it seems like thermald ignore all changes in thermal-conf.xml. Yes it write errors messages if I deliberately spoil path to the sensor, but in general its actions always the same regardless of the config content.

A typical scenario:
1. I'm run thermald service
2. I'm start watching any youtube video(my browser use software rendering, so CPU starts heating)
3. When the processor is heated to a temperature of 65-70c, thermald disables turbo boost and runs kidle_injects on all 4 threads. These 4 injections are displayed in the htop as processes that use 65-70% of the processor's resources. Injections keeps running until the CPU temperature drops to 59c. At this moment all applications lags and I can't normally use computer. If I would continue to use computer(even just surfing internet), the processor will continue to heat up and the completion of injections will be delayed.

Is there any other way to cooling my CPU, besides throttling? How I can control throttling(reduce its impact)? Why my config don't change behavior of thermald?

My setup:
```
# find /sys/class/hwmon/hwmon0/temp?_label -print -exec cat {} \;
/sys/class/hwmon/hwmon0/temp1_label
Package id 0
/sys/class/hwmon/hwmon0/temp2_label
Core 0
/sys/class/hwmon/hwmon0/temp3_label
Core 1

# find /sys/class/thermal/thermal_zone?/type -print -exec cat {} \;
/sys/class/thermal/thermal_zone0/type
INT3400 Thermal
/sys/class/thermal/thermal_zone1/type
pch_skylake
/sys/class/thermal/thermal_zone2/type
B0D4
/sys/class/thermal/thermal_zone3/type
x86_pkg_temp

# find /sys/class/thermal/cooling_device?/type -print -exec cat {} \;
/sys/class/thermal/cooling_device0/type
Processor
/sys/class/thermal/cooling_device1/type
Processor
/sys/class/thermal/cooling_device2/type
Processor
/sys/class/thermal/cooling_device3/type
Processor
/sys/class/thermal/cooling_device4/type
intel_powerclamp
/sys/class/thermal/cooling_device5/type
ath10k_thermal
```

[My config and debug-logs](https://gist.github.com/hxss/96a2c3ba8c20bed0a23f1699eed0f403). This is the test config, I was just trying to reduce the impact of throttling or to postpone its intervention to higher temperatures.

I think my problem is similar to #105.

**Comment by ArchangeGabriel** [comment]
What are you trying to achieve with thermald? It is designed to throttle the CPU in order to keep the laptop below a certain temperature. If you want fans to come into play for instance, you should not use thermald AFAIU.

And yes, the config file is mostly ignored for modern processors AFAIK.

**Comment by hxss** [comment]
I just want to keep the processor temperature within 70c. For example, when I use this laptop on windows, temperature rarely rises above 68c, even when I watch YouTube or play games. And I did not notice that windows used throttling.I'm wondering can I specify in the config with what temperature should trotting come in? Can I limit the impact of trotting? P.S.: There is no fans in my laptop. 13.01.2018, 02:24, "Bruno Pagani" <notifications@github.com>:What are you trying to achieve with thermald? It is designed to throttle the CPU in order to keep the laptop below a certain temperature. If you want fans to come into play for instance, you should not use thermald AFAIU.And yes, the config file is mostly ignored for modern processors AFAIK.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by ArchangeGabriel** [comment]
Yes, you should be able to control at which temperature throttling happen.

Regarding video watching, I suspect that your Linux isn’t set up correctly for using hardware decoding. But not sure if this is driver fault, browser fault or anything else. Generally speaking Linux requires more configuration than Windows to achieve the same level of performances (except for networking).

**Comment by hxss** [comment]
> Yes, you should be able to control at which temperature throttling happen
And how I can do this? What wrong with my config?

**Comment by ArchangeGabriel** [comment]
I don’t know, you will have to wait for @spandruvada to tell you.

**Comment by spandruvada** [comment]
On Fri, 2018-01-12 at 12:19 -0800, hxss wrote:
> Hello.
> I'm use thermald on Arch Linux(4.14.13-1-zen). My laptop - Lenovo
> Yoga 710-11isk with Intel® Core™ m5-6Y54 Processor(only passive
> cooling). I was trying to configure thermald for comfortable Linux
> usage on this laptop, but it seems like thermald ignore all changes
> in thermal-conf.xml. Yes it write errors messages if I deliberately
> spoil path to the sensor, but in general its actions always the same
> regardless of the config content.

Your system should have an manufacturer defined thermal configuration
file. This will be in /var/run/thermald/*.auto. So send a log without
passing --config-file /etc/thermald/thermal-conf.xml option.
Also attach this /var/run/thermald/*.auto.

Based on your log your system should throttle starting at 63C.

ZONE DUMP BEGIN

Zone 2: B0D4, Active:1 Bind:1 Sensor_cnt:1
..sensors.. 
sensor index:2 B0D4 /sys/class/thermal/thermal_zone2/ Async:0 
..trips.. 
index 4: type:active temp:58000 hyst:1000 zone id:2 sensor id:2 cdev
size:0
index 2: type:passive temp:63000 hyst:1000 zone id:2 sensor id:2 cdev
size:4
...
...


If you want to increase this. For example to increase to 70C. You can
send

# dbus-send --system --dest=org.freedesktop.thermald
/org/freedesktop/thermald
org.freedesktop.thermald.SetUserPassiveTemperature
string:B0D4 uint32:70000

'
Thanks,
Srinivas

**Comment by hxss** [comment]
@spandruvada [Thermald-conf.auto, debug.log & debug-70.log](https://gist.github.com/hxss/6a811253814b76de574a4ee7c6491946)
It seems that this dbus command did not affect the thermald.

Is it possible to reduce the influence of trotting(htop shows that kidle_ijects use 65-70% of the processor's resources)?

Is there any other way to cooling this mobile CPU, besides throttling? Yesterday I noticed that for several hours of work in the Windows the processor did not overclocked above 2400MHz.
![win-cpu-usage](https://user-images.githubusercontent.com/17298149/34908417-48d9e942-f8b1-11e7-955f-da11dab6b608.png)
In Linux I have max_freq=1.1GHz in powersave mode and 2.7GHz with enabled turbo. In powersave mode temperature grows very slowly. With enabled turbo mode it can go up to 80C in a few seconds.

What do "Processor" cooling devices do(/sys/class/thermal/cooling_device[0-3])?

**Comment by spandruvada** [comment]
I didn't see any dbus message sent in logs. Did you send as root?

Also there is a ThermalMonitor GUI, where you can do all in GUI, Not sure Arch Linux has this package.

Try this:

In file
/var/run/thermald/thermal-conf.xml.auto

Under SensorType BOD4, there is a 
<Temperture>*</Temperature>

Change this * to 70000  for 70C.

Linux is aggressive to ramp up compared to Windows.

Also what is the version of Thermald

#thermald --version

Thanks,
Srinivas



________________________________________

**Comment by hxss** [comment]
```
# thermald --version
1.7.1
```

>I didn't see any dbus message sent in logs

Sorry, I send this message before running thermald.

I was trying again and I updated [debug-70.log](https://gist.github.com/hxss/6a811253814b76de574a4ee7c6491946). It's very strange. I was trying to warm up CPU to 70C(for throttling activate), but temperature didn't grows above 66C. Even when I opened 4k vp9 video(my GPU doesn't support VP9 decoding). I even turned on turbo boost(I use tlp for processor controlling).
Now, after thermald was stopped I still feel small freezes of apps. As if the CPU is not operating at full capacity(but not like with throttling). And the temperature still does not grow above 66C.

As far as I understand, thermald used other mechanism than throttling(I didn't see any kidle_inject)? What is that was? Is possible to raise the allowed temperature for better performance?

**Comment by spandruvada** [comment]
First check that you are not using "performance" governor. What is
# cpupower frequency-info

Change to powersave
# cpupower frequency-set -g powersave

It is possible that you have some app which is causing CPU not to sleep. You may not have to change the threshold to 70C.

For bumping up CPU temperature using something like
# stress -c4

Thanks,
Srinivas

________________________________________

**Comment by hxss** [comment]
>First check that you are not using "performance" governor

I configured [tlp](https://wiki.archlinux.org/index.php/TLP) for always use powersave + turbo enabled([tlp-stat added](https://gist.github.com/hxss/6a811253814b76de574a4ee7c6491946)).
[debug-70.log](https://gist.github.com/hxss/6a811253814b76de574a4ee7c6491946) updated. Now I was use [s-tui](https://github.com/amanusk/s-tui). From the start of stress test temperature growed to ~80C and throttling was activate. CPU freq quickly dropped to 400 and keeps at this level even after stopping the thermald.
I don't know how to return CPU in normal mode.
![screenshot from 2018-01-15 04-44-57](https://user-images.githubusercontent.com/17298149/34922273-97be92b6-f9af-11e7-97e5-ae8d569618b1.png)

**Comment by spandruvada** [comment]
On Sun, 2018-01-14 at 23:54 +0000, hxss wrote:
> First check that you are not using "performance" governor
> I configured tlp for always use powersave + turbo enabled(tlp-stat
> added).
> debug-70.log updated. Now I was use s-tui. From the start of stress
> test temperature growed to ~80C and throttling was activate. CPU freq
> quickly dropped to 400 and keeps at this level even after stopping
> the thermald.
> I don't know how to return CPU in normal mode.
You shouldn't stop thermald once throttling is active.
It will remove all throttling slowly. If you kill thermald then you
will leave in throttled state.

Thanks,
Srinivas

**Comment by hxss** [comment]
Hello @spandruvada. I'm sorry I did not answer here. I tested config with 70000 as you advice me. In general, this is what I need. The temperature is kept within 70C and thermald doesn't use throttling, only RAPL. I think, I would use it always.
I begin to understand what is written in the debug-logs. And I have 2 questions:
1. Can I configure target state for cooling device?
`index 5: type:passive temp:70 hyst:0 zone id:3 sensor id:3 cdev size:1
cdev[0] rapl_controller target_state:not defined`
2. What is the "Processor" cooling device? I were trying to use it([debug-70-proc.log & config](https://gist.github.com/hxss/f9aee7b24f78e89264989a046af8ea5c)), but it wasn't added into dumped zone.
